### PR TITLE
OOP relationships

### PR DESCRIPTION
**Pull request by c00p75**
# OOP school library: set up associations

- Create ```class Classroom``` with accessible ```@label``` instance variable upon initialization.

- Create has-many/belongs-to relationship between ```class Classroom``` and ```class Student```.
   - ``` class Classroom``` has many students.
   - ```class Student``` belongs to a classroom.
   -  Update ```class Classroom.students``` when ```class Student.classroom``` is added/updated.
   - Update ```class Student.classroom``` when ```class Classroom.students``` is updated.
 
- Create ```class Book``` with accessible ```@title``` and ```@author``` instance variables upon initialization.

- Create ```class Rental``` with accessible ```@date``` instance variable upon initialization.

- Create many-to-many relationship between ```class Person``` and ```class Book``` using the intermediate ```class Rental```.
   - ```class Book``` has many rentals.
   - ```class Rental``` belongs to a book.
   - ```class Person``` has many rentals.
   - ```class Rental``` belongs to a person.
   - Modify ```class Rental``` constructor so ```@book``` and ```@person``` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (nahnahdev) [inline book.rb]: - [x] Great job so far @c00p75 you're making good progress. 🥳  I wanted to provide some feedback on the implementation of the `many-to-many relationship` between `Person` and `Book` using the intermediate class `Rental`.

- [x] In your Book class, you've correctly defined the rentals array to store the associated rentals. However, to fully meet the requirements, you need to implement the logic to associate Rental instances with Book objects. 💯 

- [x] I recommend adding a method called `add_rental` in your `Book class `that allows you to associate rentals with a book 👍 By implementing this logic, you'll fully establish the `has-many relationship` between `Book` and `Rental`, fulfilling the requirements of the project 💯 
- Project Requirement

> Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
Create the has-many side of Book and Rental (a book has many rentals).

File: person.rb [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
   def initialize(age, name = 'unknown', parent_permission: true)
     @age = age
     @name = name
     @parent_permission = parent_permission
     @id = Random.rand(1..100)
+    @rentals = []
+    super()
   end
 
   def can_use_services?
```
Comment (nahnahdev) [inline person.rb:16]: - [x] In your Person class, I noticed that you've correctly defined the rentals array to store the associated rentals. However, to fully meet the requirements, you need to implement the logic to associate Rental instances with Person objects. 👍 

- [x] I recommend adding a method called `add_rental` in your Person class that allows you to associate rentals with a person just as I suggested earlier in the Book class 😉 
- By implementing this logic, you'll fully establish the `has-many relationship` between `Person` and `Rental`, fulfilling the requirements of the project 💯 
- Project Requirement

> Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
> Create the has-many side of Person and Rental (a person has many rentals).

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @c00p75 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters well setup and passing ✔️ 
- Good use of GitFlow ✔️ 
- Professional README ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @c00p75 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters well setup and passing ✔️ 
- Good use of GitFlow ✔️ 
- Professional README ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (c00p75) [reply person.rb:16]: Well noted!💯
Thanks for the thorough and clear feedback 💪

**Review by Zuhaib042 — APPROVED** [review]
Hi @c00p75 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Descriptive PR
✅ Professional Readme
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**